src/foam/dao/FixedSizeDAO.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package foam.dao;
+
+import java.util.Iterator;
+import foam.core.*;
+import foam.util.ConcurrentList;
+import foam.util.List;
+import foam.mlang.order.Comparator;
+import foam.mlang.predicate.Predicate;
+
+public class FixedSizeDAO
+  extends AbstractDAO
+{
+  private List<FObject> list;
```

Comment:
Make instance variables be protected instead of private and end names with underscores. ie. rename list to list_.